src/main/java/org/assertj/core/internal/ComparatorBasedComparisonStrategy.java

```diff
@@ -94,6 +95,10 @@ public boolean iterableContains(Iterable<?> iterable, Object value) {
   @SuppressWarnings("unchecked")
   public void iterableRemoves(Iterable<?> iterable, Object value) {
     if (iterable == null) return;
+    // Avoid O(N^2) complexity of serial removal from an iterator of collections like ArrayList
```

Comment:
Unless I'm missing something the original removal is O(n) since it is just iterating over the elements. do you agree?

Reply:
Each removal is O(N), and there are O(N) of them